vpc/service/action_worker.go

```diff
@@ -122,10 +127,16 @@ func (actionWorker *actionWorker) loop(ctx context.Context, item keyedItem) erro
 					err = errors.Wrap(err, "Could not retrieve all work items")
 					return err
 				}
-				if !actionWorker.readyChClosed {
-					close(actionWorker.readyCh)
-					actionWorker.readyChClosed = true
-				}
+				actionWorker.readyCond.L.Lock()
+				actionWorker.ready = true
+				actionWorker.readyCond.L.Unlock()
```

Comment:
I don't really understand, shouldn't this stay Locked? Why unlock right away?

Reply:
I added a comment in the struct trying to make this make more sense.